combat-trainer.lic

```diff
@@ -1,3344 +1,3554 @@
-    end
-  end
-
-  def check_nonspell_buffs(game_state)
-    @buffs.each do |action, cooldown|
-      timer = game_state.cooldown_timers[action]
-      next unless !timer || (Time.now - timer).to_i > cooldown
-      game_state.cooldown_timers[action] = Time.now
-      fput action
-      waitrt?
-    end
-    return if game_state.danger && @kneel_khri
-    @khri
-      .map { |name| "Khri #{name}" }
-      .each do |name|
-        timer = game_state.cooldown_timers[name]
-        next unless !timer || (Time.now - timer).to_i > 30
-        game_state.cooldown_timers[name] = Time.now if activate_khri?(@kneel_khri, name)
-      end
-    @barb_buffs
-      .select { |name| Flags["ap-#{name}-expired"] }
-      .each do |name|
-        Flags.reset("ap-#{name}-expired")
-        activate_barb_buff(name)
-        echo "Activated barb_buff #{name}!" if $debug_mode_ct
-        waitrt?
-        pause 6 if get_data('spells').barb_abilities[name]['type'].eql? 'meditation'
-      end
-  end
-end
-
-class ManipulateProcess
-  include DRC
-  include DRCT
-
-  def initialize(settings)
-    echo('New ManipulateProcess') if $debug_mode_ct
-
-    @threshold = settings.manipulate_threshold
-    echo("  @threshold: #{@threshold}") if $debug_mode_ct
-
-    @manip_to_train = settings.manipulate_only_to_train
-    echo("  @manip_to_train: #{@manip_to_train}") if $debug_mode_ct
-
-    @last_manip = Time.now - 200
-  end
-
-  def execute(game_state)
-    return if game_state.danger || @threshold.nil? || game_state.construct_mode?
-    @filtered_npcs = game_state.npcs
-    manipulate(game_state) if should_manipulate?
-  end
-
-  private
-
-  def should_manipulate?
-    return false if @manip_to_train && DRSkill.getxp('Empathy') > 30
-
-    @filtered_npcs.length >= @threshold && Time.now - @last_manip > 120
-  end
-
-  def manipulate(game_state)
-    bput('manip stop all', 'You relax your will', 'But you aren')
-
-    manipulate_count = 0
-    npc_occurrences = Hash.new(0)
-
-    @filtered_npcs.each do |npc|
-      next if game_state.construct?(npc)
-      break if manipulate_count >= @threshold
-
-      index = npc_occurrences[npc]
-      ordinal_string = $ORDINALS[index]
-
-      case bput("manipulate friendship #{ordinal_string} #{npc}", 'You\'re already manipulating', 'beyond your ken', 'You attempt to empathically manipulate', 'You strain', 'does not seem to have a life essence')
-      when 'does not seem to have a life essence' then
-        game_state.construct(npc)
-      else
-        manipulate_count += 1
-        npc_occurrences[npc] += 1
-      end
-    end
-
-    @last_manip = Time.now
-  end
-end
-
-class TrainerProcess
-  include DRC
-  include DRCT
-
-  def initialize(settings, equipment_manager)
-    @equipment_manager = equipment_manager
-    echo('New TrainerProcess') if $debug_mode_ct
-    @training_abilities = settings.training_abilities
-    ech
```

Comment:
❓ Is a lot of this a duplicate of what is already in `common-arcana`?

Reply:
Looking at it there is there differences here and there??  Maybe some could be added to common-arcana?  Right now the defined things I see are to help moonwatch run.  @jandersson  what do you think since I stole it from astrology?